distributed/diagnostics/plugin.py

```diff
@@ -96,6 +94,14 @@ def transition(
     ) -> None:
         """Run whenever a task changes state
 
+        For a description of the transition mechanism and the available states,
+        see :ref:`Scheduler task states <scheduler-task-state>`.
+
+        Note::
```

Comment:
Maybe this should be a warning?
```suggestion
        Warning::
```

Reply:
yeah, I was not sure how much I wanted to scare people off... I personally would prefer for this API not even to exist so I guess a warning is indeed more appropriate :)